kinesis/src/main/java/io/debezium/examples/kinesis/ChangeDataSender.java

```diff
@@ -85,20 +87,39 @@ public void run() {
             engine.stop();
         }));
 
+        // the submitted task keeps running, only no more new ones can be added
+        executor.shutdown();
+
         awaitTermination(executor);
+
+        cleanUp();
+
+        LOGGER.info("Engine terminated");
     }
 
     private void awaitTermination(ExecutorService executor) {
         try {
             while (!executor.awaitTermination(10, TimeUnit.SECONDS)) {
-                LOGGER.info("Waiting another 10 seconds for the embedded engine to shut down");
+                LOGGER.info("Waiting another 10 seconds for the embedded engine to complete");
             }
         }
         catch (InterruptedException e) {
-            Thread.interrupted();
+            Thread.currentThread().interrupt();
         }
     }
 
+    private void cleanUp() {
+        // Lingering MySQL thread without this...
+        try {
+            AbandonedConnectionCleanupThread.shutdown();
```

Comment:
Why is this needed? The thread in question is daemon thread. If the thread does not stop then there must be another thread lingering. Also the method in question is deprecated.

Reply:
@jpechane, try to remove it, the thread will stick around. Happy to change it, if there's better ways to do it.

Reply:
@gunnarmorling I've just checked it for NATS - by the end of shutdown hook only daemon threads are alive (one of them is the MySQL dameon thread). hence I believe it is unnecessary. Otherwise you'd be required to excute the very same operation for any application that uses MySQL JDBC driver.